src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -1760,7 +1760,7 @@ int xclResetDevice(xclDeviceHandle handle, xclResetKind kind)
 int xclP2pEnable(xclDeviceHandle handle, bool enable, bool force)
 {
     xocl::shim *drv = xocl::shim::handleCheck(handle);
-    return drv ? drv->p2pEnable(enable, force) : -ENODEV;
+    return drv ? drv->p2pEnable(enable, force) : ENODEV;
```

Comment:
hmm...why?

Reply:
strerror() doesn't take -ve value in xclP2p xbutil.cpp. But I think it is better to change "int ret" to unsigned instead of this change.

Reply:
it's more consistent to return negative value in similar functions here. we should pass -val to strerror().